tools/language-server/logstat/src/main.rs

```diff
@@ -0,0 +1,322 @@
+    };
+}
+
+/// Read the file with specification
+async fn read_specs(path: &PathBuf) -> Result<Spec> {
+    let file = File::open(path).await?;
+    let lines_reader = BufReader::new(file).lines();
+    let matches = LinesStream::new(lines_reader).collect::<Result<Vec<_>>>().await?;
+
+    Ok(Spec { matches })
+}
+
+/// Extract the iterations information from logfile according to the provided spec
+async fn read_logfile(path: &PathBuf, spec: &Spec) -> Result<Vec<Iteration>> {
+    let file = File::open(path).await?;
+    let mut lines = BufReader::new(file).lines();
+
+    let mut iterations = vec![];
+    let mut current_operations = vec![];
+    let mut matches = spec.matches.iter();
+    let mut current_match = matches.next().expect("Empty spec!");
+
+    while let Some(line) = lines.next_line().await? {
+        if let Some(cap) = RE_LOGLINE.captures(line.as_str()) {
+            let groups = (
+                cap.get(RE_LOGLINE_TIMESTAMP_CAPTURE_GROUP),
+                cap.get(RE_LOGLINE_MESSAGE_CAPTURE_GROUP),
+            );
+            match groups {
+                (Some(timestamp), Some(message)) =>
+                    if message.as_str().contains(current_match) {
+                        let duration = Duration::ZERO;
+                        let timestamp =
+                            OffsetDateTime::parse(timestamp.as_str(), &Rfc3339).unwrap();
+                        let line = message.as_str().to_string();
+
+                        current_operations.push(Operation { duration, timestamp, line });
+
+                        if let Some(m) = matches.next() {
+                            current_match = m
+                        } else {
+                            matches = spec.matches.iter();
+                            current_match = matches.next().expect("Empty spec!");
+                            iterations.push(Iteration { operations: current_operations });
+                            current_operations = vec![];
+                        }
+                    },
+                _ => {
+                    eprintln!("[ERR] Invalid log line [{}]", line);
+                }
+            }
+        }
+    }
+
+    Ok(iterations)
+}
+
+/// Calcualte median of values
+fn median<I>(durations_iter: I) -> Duration
+where I: Iterator<Item = Duration> {
+    let mut durations = durations_iter.collect::<Vec<_>>();
+    durations.sort();
+
+    let mid = durations.len() / 2;
+    if durations.len() % 2 == 0 {
+        (durations[mid - 1] + durations[mid]) / 2
+    } else {
+        durations[mid]
+    }
+}
+
+/// Cleanup iterations info before analyzing
+fn cleanse_iterations(ops: &mut Vec<Iteration>, skip: usize) {
+    ops.drain(..skip);
+}
+
+/// Merge iterations from two log files
+fn merge_iterations(ws_iterations: &mut [Iteration], log_iterations: Vec<Iteration>) {
+    for (log_iter, ws_iter) in log_iterations.into_iter().zip(ws_iterations.iter_mut()) {
+        ws_iter.operations.splice(1..1, log_iter.operations);
```

Comment:
Why do we insert operations just at _second_ position? Can it be explained by comment or by making "1" a well-named constant?